sql/sqlite/inspect_test.go

```diff
@@ -307,14 +314,100 @@ CREATE TABLE t1 (
 		},
 	}
 	for _, tt := range tests {
-		m := reConstC.FindStringSubmatch(tt.input)
+		m := reFKC.FindStringSubmatch(tt.input)
 		require.Equal(t, len(m) != 0, len(tt.matches) != 0)
 		if len(m) > 0 {
 			require.Equal(t, tt.matches, m[1:])
 		}
 	}
 }
 
+func TestRegex_Checks(t *testing.T) {
+	tests := []struct {
+		input  string
+		checks []*schema.Check
+	}{
+		{
+			input: `CREATE TABLE pets (id int NOT NULL, owner_id int CONSTRAINT "ck1" CHECK (owner_id <> 0))`,
+			checks: []*schema.Check{
+				{Name: "ck1", Expr: "(owner_id <> 0)"},
+			},
+		},
+		{
+			input: `CREATE TABLE pets (id int NOT NULL, owner_id int CHECK (owner_id <> 0) CONSTRAINT "ck1")`,
+			checks: []*schema.Check{
+				{Expr: "(owner_id <> 0)"},
+			},
+		},
+		{
+			input: `CREATE TABLE pets (id int NOT NULL CHECK ("id" <> 0), owner_id int CONSTRAINT "ck1" CHECK ((owner_id) <> 0))`,
+			checks: []*schema.Check{
+				{Expr: `("id" <> 0)`},
+				{Name: "ck1", Expr: "((owner_id) <> 0)"},
+			},
+		},
+		{
+			input: `CREATE TABLE pets (id int NOT NULL CHECK ("(" <> ')'), owner_id int CONSTRAINT "ck1" CHECK ((owner_id) <> 0))`,
+			checks: []*schema.Check{
+				{Expr: `("(" <> ')')`},
+				{Name: "ck1", Expr: "((owner_id) <> 0)"},
+			},
+		},
+		{
+			input: "CREATE TABLE pets (\n\tid int NOT NULL CHECK (id <> 0) CHECK ((id % 2) = 0)\n,\n\towner_id int CHECK ((owner_id) <> 0)\n)",
+			checks: []*schema.Check{
+				{Expr: "(id <> 0)"},
+				{Expr: "((id % 2) = 0)"},
+				{Expr: "((owner_id) <> 0)"},
+			},
+		},
+		{
+			input: `CREATE TABLE t1(
+				x INTEGER CHECK( x<5 ),
+				y REAL CHECK( y>x ))`,
+			checks: []*schema.Check{
+				{Expr: "( x<5 )"},
+				{Expr: "( y>x )"},
+			},
+		},
+		{
+			input: `CREATE TABLE t(
+				x INTEGER CONSTRAINT one CHECK( typeof(coalesce(x,0))=="integer" ),
+				y NUMERIC CONSTRAINT two CHECK( typeof(coalesce(y,0.1))=='real' ),
+				z TEXT CONSTRAINT three CHECK( typeof(coalesce(z,''))=='text' )
+			)`,
+			checks: []*schema.Check{
+				{Name: "one", Expr: `( typeof(coalesce(x,0))=="integer" )`},
+				{Name: "two", Expr: `( typeof(coalesce(y,0.1))=='real' )`},
+				{Name: "three", Expr: `( typeof(coalesce(z,''))=='text' )`},
+			},
+		},
+		{
+			input: `CREATE TABLE t(
+				x char check ('foo''(' <> 1)
+			)`,
+			checks: []*schema.Check{
+				{Expr: `('foo''(' <> 1)`},
+			},
+		},
+		// Invalid inputs.
+		{
+			input: "CREATE TABLE t(x char check)",
+		},
+		{
+			input: "CREATE TABLE t(x char constraint x check)",
+		},
+	}
+	for _, tt := range tests {
+		table := &schema.Table{Attrs: []schema.Attr{&CreateStmt{S: tt.input}}}
+		require.NoError(t, fillChecks(table))
```

Comment:
In general, I think unit tests should test public methods. Is there a way to check this logic without directly to an unexported method?

Reply:
It was @yonidavidson request the last time I’ve added a regex.